### PR TITLE
Fixed issue #19331: Try to add a subquestion to a matrix type question throws an SQL error

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

**Review by Shnoulle — APPROVED** [review]

**Comment by Shnoulle** [comment]
Test done (on pgsql) by reporter : https://bugs.limesurvey.org/view.php?id=19332

**Comment by Shnoulle** [comment]
@gabrieljenik with php8.3

`DOMAIN=lstest.local  phpunit ./tests/unit/services/QuestionAggregateService/SubquestionsServiceTest.php`
```
…
	<p class="message">
		Calling get_class() without arguments is deprecated	</p>

	<div class="source">
		<p class="file">/media/shnoulle/data/webdev/lstest/application/helpers/ClassFactory.php(39)</p>

```

**Comment by Shnoulle** [comment]
Move the error from 8.2 to 7.4 ??????  WTF ?

File: application/models/services/QuestionAggregateService/SubQuestionsService.php [hunk 1]
```diff
@@ -162,6 +162,9 @@ private function storeSubquestion(
                 $subquestion = DI::getContainer()->make(Question::class);
             }
         }
+        if (empty($subquestion)) {
```
Comment (Shnoulle) [inline application/models/services/QuestionAggregateService/SubQuestionsService.php:164]: I think we must avoid this before ?

No ?

Throw an exception is a really good idea :)

Comment (gabrieljenik) [reply application/models/services/QuestionAggregateService/SubQuestionsService.php:164]: > I think we must avoid this before 

Not sure there are lot of places before where to move this... do you find a better spot?

Comment (Shnoulle) [reply application/models/services/QuestionAggregateService/SubQuestionsService.php:164]: > > I think we must avoid this before
> 
> Not sure there are lot of places before where to move this... do you find a better spot?

Sorry, I didn't make myself clear :)

I mean : call SubQuestionsService with invalid values for sub qids must be checked before :) and have an error management before.

Here : seems a good place inside service, but my opinion : this is more a error before (in controller) : value send to service must be clean.

**Pull request merged**